### DISCUSSION TITLE
Issue retrieving the ContentItemId workflow from Input dictionary

### DISCUSSION THREAD

**Discussion by pbull001** [OrchardCMS/OrchardCore#16136] [Q&A]
I am very confused about retrieving the ContentItemId of the item the subject of a ContentUpdated event. I created a workflow setting three properties (name:expression) ContentItem0 : input("ContentItem); , ContentItem: input("ContentItem).ContentId; and input("ContentEvent").ContentItemId; My questions is why does the 2nd expression return with null? 

![image](https://github.com/OrchardCMS/OrchardCore/assets/18185015/972b5883-f158-4fc5-a770-fc7b27f75e20)

**Comment by douwinga** [comment]
Did you see my reply in Discord? `input("ContentItem)` should actually be the full content item, the stored state is just misleading. Not sure the reasoning for showing the state like that is, but you should be able to access any of the content item properties from `input("ContentItem")`. So `input("ContentItem").ContentItemId` should get you the correct value you are after but you also have access to things like the content parts via something like `input("ContentItem").Content.MyPart.MyField.Value`.

**Reply by pbull001** [reply]
I did see your response. Thank you. I was asked to turn this into a Question here. I am still curious as to why second expression returns null?

**Reply by douwinga** [reply]
So `input("ContentEvent").ContentItemId` returns null? I am not sure the ContentEvent is supposed to accessible, and I have never tried to use it. I am not seeing a `TemplateOptions` registration for its model `ContentEventContext`, so I don't think it will be as simple as `input("ContentEvent").ContentItemId` to get its value. I think if for some reason you really needed these values, you would have to do something gross like
```javascript
var contentEventJson = JSON.stringify(input("ContentEvent"))
var contentEvent = JSON.parse(contentEventJson)
```
and then you would have `contentEvent.ContentItemId`. I have not tested this though, and I think since you already have the content item, this would be completely unnecessary.
